### PR TITLE
Stl 2050 ttl capability

### PR DESCRIPTION
**Pull request by VictorDelCampo**
# Short Description
- TTL for the queueing jobs in the redis
- Cron-jobs for the entities in the postgres

# Changes
- Added three new variable in the config
- Added new ttl property when creating the job
- Created cron job for the postgres
- Added docker-compose.ci.yml (targets for the erica api and the cron)
- Adapted dockerfile
- Adapted pipeline

# Feedback
- (RESOLVED) I tried it locally (docker-compose -f docker-compose.ci.yml build) by setting manually the env args and it worked. But I needed to set the "elster_hersteller_id" with an extra $ so that an interpolation error wouldn't come up (because of the already present $ character) See https://docs.docker.com/compose/compose-file/compose-file-v2/#variable-substitution. That means with need to add an extra $ at the beginning of the env variable <-- **this is not a problem anymore since I moved back the ENV variables to the pipeline file**

# How to review this Pull Request
[Link to Confluence](https://digitalservice4germany.atlassian.net/wiki/spaces/STL/pages/117637157/Development+Process#Approve-or-comment) (internal)

### CODE REVIEW

**Review by JulianRoesner — COMMENTED** [review]
In general, it looks very good. Let's maybe quickly talk about the comments.

File: erica/application/EricaRequest/EricaRequestService.py [hunk 1]
```diff
@@ -23,8 +23,8 @@ def get_all_by_skip_and_limit(self, skip: int, limit: int):
 
 
 class EricaRequestService(EricaRequestServiceInterface):
-
     erica_request_repository: EricaRequestRepository
+    scheduler = None
```
[the remark this answers is not in the document]
Comment (VictorDelCampo) [reply erica/application/EricaRequest/EricaRequestService.py:27]: Forgot to remove it from the previous implementation!

File: erica/infrastructure/sqlalchemy/cron/jobs.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import logging
```
[the remark this answers is not in the document]
Comment (VictorDelCampo) [reply erica/infrastructure/sqlalchemy/cron/jobs.py:1]: Alright! Done!

File: erica/config.py [hunk 3]
```diff
@@ -29,6 +29,9 @@ class Settings(BaseSettings):
     default_queues: str = 'dongle, cert, common'
     database_url: str = Field("postgresql://postgres:postgres@localhost/db", env="ERICA_DATABASE_URL")
     dongle_connected: bool = Field(True, env="DONGLE_CONNECTED")
+    ttl_queuing_job_in_sec: int = 60
+    ttl_processing_entities_in_min: int = 2
+    ttl_success_fail_entities_in_min: int = 10
```
[the remark this answers is not in the document]
Comment (VictorDelCampo) [reply erica/config.py:34]: Done!

File: docker-compose.ci.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# Note: this docker-compose file is used for building staging/production images on GitHub Actions.
+version: '3.8'
+
+services:
+  web:
+    image: erica_app:${DOCKER_TAG:-latest}
```
[the remark this answers is not in the document]
Comment (VictorDelCampo) [reply docker-compose.ci.yml:6]: Changed back to erica_api

File: tests/infrastructure/sqlalechemy/repositories/mock_repositories.py [hunk 5]
```diff
@@ -22,3 +29,7 @@ class MockSchema(BaseDbSchema):
                 primary_key=True)
     request_id = Column(UUID(as_uuid=True))
     payload = Column(JSONB)
+    status = Column(Enum(Status))
+    updated_at = Column(sqlalchemy.types.DateTime(timezone=True))
```
[the remark this answers is not in the document]
Comment (VictorDelCampo) [reply tests/infrastructure/sqlalechemy/repositories/mock_repositories.py]: Removed

**Comment by VictorDelCampo** [comment]
@SannyNguyenHung reminder here :)

**Review by SannyNguyenHung — APPROVED** [review]
LGTM! Thank you for the new tests for the repository :+1:

**Review by SannyNguyenHung — APPROVED** [review]
Wow didn't know "x-" is a thing in docker :smile:

**Pull request merged**